src/objects/zcl_abapgit_object_aifc.clas.abap

```diff
@@ -0,0 +1,509 @@
+      RAISING
+        zcx_abapgit_exception.
+    METHODS get_content_compress
+      IMPORTING
+        !io_log     TYPE REF TO zif_abapgit_log
+        !is_ifkeys  TYPE ty_aif_key_s
+        !iv_package TYPE devclass
+      RAISING
+        zcx_abapgit_exception.
+    METHODS validate_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception.
+    METHODS compress_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+        !io_log           TYPE REF TO zif_abapgit_log
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception.
+
+  PRIVATE SECTION.
+    TYPES:
+      BEGIN OF ty_content_s,
+        tabname TYPE tabname,
+      END OF ty_content_s.
+    TYPES:
+      ty_content_t TYPE STANDARD TABLE OF ty_content_s WITH NON-UNIQUE DEFAULT KEY.
+
+    DATA mo_abapgit_util TYPE REF TO object.
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
+
+
+  METHOD authorization_check.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    rv_success = abap_false.
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~AUTHORIZATION_CHECK')
+          RECEIVING
+            rv_success = rv_success.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lx_dyn_call_error ).
+      CATCH cx_root INTO lx_root.
+        io_log->add_exception( ix_exc  = lx_root
+                               is_item = ms_item ).
+        RETURN.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD compress_interface.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~COMPRESS_INTERFACE')
+          EXPORTING
+            is_ifkeys  = is_ifkeys
+          RECEIVING
+            rv_success = rv_success.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lx_dyn_call_error ).
+
+      CATCH cx_root INTO lx_root.
+        io_log->add_exception( ix_exc  = lx_root
+                               is_item = ms_item ).
+        RETURN.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD constructor.
+    DATA: lx_exc_ref TYPE REF TO cx_sy_dyn_call_error.
+
+    super->constructor( is_item     = is_item
+                        iv_language = iv_language ).
+
+    CLEAR ms_icd_data_key.
+    ms_icd_data_key = is_item-obj_name.
+
+    TRY.
+        CALL METHOD ('/AIF/CL_ABAPGIT_AIFC_UTIL')=>('GET_INSTANCE')
+          RECEIVING
+            rr_abapgit_aifc_util = mo_abapgit_util.
+
+      CATCH cx_sy_dyn_call_error INTO lx_exc_ref.
+    ENDTRY.
```

Comment:
When I try to test the demo repo in my system I get this shotdump because it doesn't support AIFC objects.
![grafik](https://user-images.githubusercontent.com/17437789/146135812-f58a8de7-c687-4801-8ab2-3d13b59dd7ba.png)

We can fix it like this:
```suggestion
      CATCH cx_sy_dyn_call_error INTO lx_exc_ref.
          zcx_abapgit_exception=>raise( 'AIFC not supported' ).
    ENDTRY.
```
and add `raising zcx_abapgit_exception` to `CONSTRUCTOR` signature.